### ISSUE TITLE
steadystate() cannot handle Liouvillians with non-unique steady states

### ISSUE THREAD

**Issue by jonahpeter** [qutip/qutip#2175]
### Bug Description

The function steadystate() does not seem capable of handling Liouvillians with non-unique steady states. This is immediately apparent because steadystate() does not contain an argument for an initial state at time zero. Instead, it returns a state (presumably some linear combination of the null-space operators) that does not match the long time output from mesolve(), which varies based on initial condition.

There are many Liouvillians which do not admit unique steady states, but a simple example can be found in Section 3.3 of [this tutorial](https://doi.org/10.48550/arXiv.2303.16449).

### Code to Reproduce the Bug

```shell
The bug is apparent from the documentation, as steadystate() does not contain an argument for an initial state.
```

### Expected Behaviour

For Liouvillians that admit non-unique steady states, I expected steadystate() to return the long time limit of mesolve() given a particular initial condition.

### Your Environment

```shell
QuTiP: Quantum Toolbox in Python
================================
Copyright (c) QuTiP team 2011 and later.
Current admin team: Alexander Pitchford, Nathan Shammah, Shahnawaz Ahmed, Neill Lambert, Eric Giguère, Boxi Li, Jake Lishman and Simon Cross.
Board members: Daniel Burgarth, Robert Johansson, Anton F. Kockum, Franco Nori and Will Zeng.
Original developers: R. J. Johansson & P. D. Nation.
Previous lead developers: Chris Granade & A. Grimsmo.
Currently developed through wide collaboration. See https://github.com/qutip for details.

QuTiP Version:      4.6.2
Numpy Version:      1.21.2
Scipy Version:      1.7.1
Cython Version:     0.29.24
Matplotlib Version: 3.5.0
Python Version:     3.7.11
Number of CPUs:     2
BLAS Info:          INTEL MKL
OPENMP Installed:   False
INTEL MKL Ext:      True
Platform Info:      Darwin (x86_64)
Installation path:  /opt/anaconda3/lib/python3.7/site-packages/qutip
================================================================================
Please cite QuTiP in your publication.
================================================================================
For your convenience a bibtex reference can be easily generated using `qutip.cite()`
```

**Comment by nwlambert** [comment]
Yeah in my experience this is something one has to be careful of with the steady-state solver as it is. Depending on the method chosen it can either fail, return one of the possibilities, or some linear combination of them.  The default one (direct) tends to fail, which at least sometimes lets you know you have this issue. 

We could consider adding null_space solver which returns all possibilities, or modify the existing svd solver to do so if possible?  
Still, if the default "direct" method is silently failing it may trip people up still, so adding some examples to the documentation might help, as a minimum.  I don't know of a way to extend that direct method to return all possible solutions, and connect those to possible initial conditions.  It would be very interesting if its possible!

One minor thing; looking at the the example in the linked paper you provided seems to badly constructed.  The dephasing operator is an identity, so does nothing, so technically the second example has no well-defined steady-state at all.  A more useful example would be a proper dephasing through a collapse operator = sigma_z,  and no driving on the qubit, so the degenerate steady-states are <sigmaz>=\pm 1.

**Comment by veronikakurth** [comment]
@Ericgig Does your PR #2508 also solve this issue?

**Comment by matti-sarjala** [comment]
A small documentation example could make this much clearer by separating two questions:

1. What is the steady-state null space of the Liouvillian?
2. Which point in that null space is selected by a particular initial state?

For a primitive Liouvillian these collapse to the same practical question, because the steady state is unique. But for a Liouvillian with multiple stationary states, `steadystate(L)` cannot generally return “the” long-time state without extra information about the initial condition.

A minimal toy model could be pure dephasing with collapse operator `sigmaz()` and no Hamiltonian drive:

- populations are conserved,
- coherences decay,
- every diagonal density matrix is stationary,
- different initial populations lead to different long-time diagonal states.

That example would show why the issue is not only numerical. The null space is genuinely multidimensional, so the API/documentation should distinguish:

- returning one representative steady state,
- returning a basis for the steady-state subspace,
- projecting an initial state onto the asymptotic steady-state manifold.

This would help users understand why `steadystate()` and long-time `mesolve()` need not agree when the steady state is not unique.
